google/cloud/forseti/services/inventory/storage.py

```diff
@@ -636,6 +761,91 @@ def _get_resource_id(self, resource):
 
         return 0
 
+    def clear_cai_data(self):
+        """Deletes all temporary CAI data from the cai temporary table.
+
+        Returns:
+            int: The number of rows deleted.
+        """
+        num_rows = CaiTemporaryStore.delete_all(self.session)
+        self.has_cai_data = False
+        return num_rows
+
+    def populate_cai_data(self, data):
+        """Add assets from cai data dump into cai temporary table.
+
+        Args:
+            data (str): Line delimeted text dump of json data representing
+                assets from Cloud Asset Inventory ExportAssets API.
+
+        Returns:
+            int: The number of rows inserted
+        """
+        num_rows = 0
+        for line in data.split('\n'):
+            if not line:
+                continue
+            row = CaiTemporaryStore.from_json(line)
+            self.buffer.add(row)
+            num_rows += 1
+        self.buffer.flush()
+        self.session.commit()
+        self.has_cai_data = True
+        return num_rows
+
+    def iter_cai_assets(self, content_type, asset_type, parent_name):
+        """Iterate the objects in the cai temporary table.
+
+        Args:
+            content_type (ContentTypes): The content type to return
+            asset_type (str): The asset type to return
+            parent_name (str): The parent resource to iter children under.
+
+        Yields:
+            object: The content_type data for each resource.
+        """
+        if not self.has_cai_data:
+            return
+
+        filters = [
+            CaiTemporaryStore.content_type == content_type,
+            CaiTemporaryStore.asset_type == asset_type,
+            CaiTemporaryStore.parent_name == parent_name,
+        ]
+        base_query = self.session.query(CaiTemporaryStore)
+
+        for qry_filter in filters:
+            base_query = base_query.filter(qry_filter)
+
+        base_query = base_query.order_by(CaiTemporaryStore.name.asc())
+
+        for row in base_query.yield_per(PER_YIELD):
+            yield row.extract_asset_data(content_type)
+
+    def fetch_cai_asset(self, content_type, asset_type, name):
+        """Returns a single resource from the cai temporary store.
+
+        Args:
+            content_type (ContentTypes): The content type to return
+            asset_type (str): The asset type to return
+            name (str): The resource to return.
+
+        Returns:
+            dict: The content data for the specified resource.
+        """
+        if not self.has_cai_data:
+            return {}
+
+        row = self.session.query(CaiTemporaryStore).filter(
```

Comment:
qq: Is there any reason why we are using 2 diff styles to do filtering (compare with above)?

> 

        filters = [
            CaiTemporaryStore.content_type == content_type,
            CaiTemporaryStore.asset_type == asset_type,
            CaiTemporaryStore.parent_name == parent_name,
        ]
        base_query = self.session.query(CaiTemporaryStore)
         for qry_filter in filters:
            base_query = base_query.filter(qry_filter)

Reply:
You're right, fixed.